plugins/logs/plugin.go

```diff
@@ -626,10 +630,30 @@ func (p *Plugin) oneShot(ctx context.Context) (ok bool, err error) {
 			err = uploadChunk(ctx, p.manager.Client(p.config.Service), p.config.PartitionName, bs)
 		}
 		if err != nil {
-			// requeue the chunk
-			p.mtx.Lock()
-			p.bufferChunk(p.buffer, bs)
-			p.mtx.Unlock()
+			if p.limiter != nil {
+				gr, gerr := gzip.NewReader(bytes.NewReader(bs))
+				if gerr != nil {
+					continue
+				}
+
+				var events []EventV1
+				if decErr := json.NewDecoder(gr).Decode(&events); decErr != nil {
+					continue
+				}
+				gr.Close()
+
+				for _, event := range events {
+					p.mtx.Lock()
```

Comment:
Nit: any reason why we're locking and unlocking on each iteration? I'd probably just lock/unlock outside the loop, so that we only incur the cost once.

Reply:
Done